clients/drcachesim/simulator/cache_misses_analyzer.h

```diff
@@ -0,0 +1,137 @@
+/* **********************************************************
+ * Copyright (c) 2015-2018 Google, LLC  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL GOOGLE, LLC OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* cache_misses_analyzer: finds the load instructions suffering from
+ * a significant number of last-level cache (LLC) misses. In addition,
+ * it analyzes the data memory addresses accessed by these load instructions
+ * and identifies patterns that can be used in SW prefetching.
+ */
+
+#ifndef _CACHE_MISSES_ANALYZER_H_
+#define _CACHE_MISSES_ANALYZER_H_ 1
+
+#include <map>
+#include <stdint.h>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "cache_simulator.h"
+#include "cache_stats.h"
+#include "../common/memref.h"
+
+// Represents the SW prefetching recommendation passed to the compiler.
+struct prefetching_recommendation_t {
+    uint64_t pc;      // Load instruction's address.
+    int64_t stride;   // Prefetching stride/delta distance.
+    std::string locality;  // Prefetching locality: one of "nta" or "t0".
```

Comment:
Should this be an enum, or should those 2 strings be static const string values?